File: Assignment 2/Lumintrial.py
import numpy as np
from matplotlib import pyplot as plt
import pandas as pd
from scipy.stats import kendalltau
from scipy.stats import spearmanr
from scipy.stats import pearsonr


a = pd.read_excel("C:\\Users\Heera Baiju\\Downloads\\Data Science\\Assignment 2\\LuminosityHeera.xlsx")
Luminosity = np.log(a["Lx"])
Redshift = np.log(a["z"])
# Resizing the figure
plt.figure(figsize=[7, 5])

# Plotting the scatter plot
plt.scatter(Luminosity, Redshift, c='g', alpha=0.6)

plt.title('Luminosity vs redshift', fontsize=15)
plt.xlabel('Luminosity', fontsize=13)
plt.ylabel('Redshift', fontsize=13)
plt.show()

Pearson_Correlation_Coefficient, p_value_p = pearsonr(Luminosity, Redshift)
Kendalltau_Correlation_Coefficient, p_value_k = kendalltau(Luminosity, Redshift)
Spearman_Correlation_Coefficient, p_value_s = spearmanr(Luminosity, Redshift)
print('Pearsons correlation: {}' .format(Pearson_Correlation_Coefficient))
print('p value of Pearson Correlation Coefficient: {}' .format(p_value_p))
print('Kendalltau correlation: {}' .format(Kendalltau_Correlation_Coefficient))
print('p value of Kendalltau Correlation Coefficient: {}' .format(p_value_k))
print('Spearman correlation: {}' .format(Spearman_Correlation_Coefficient))
print('p value of Spearman Correlation Coefficient: {}' .format(p_value_s))

File: Assignment 2/Lumintrial.r
# Importing necessary libraries
import numpy as np
from matplotlib import pyplot as plt

# Preparing the data for the plot
#x = np.arange(1, 100, 5)
#y = 32 * x

Luminosity = array('i', [345.2,
66.3,
684.0, 
209.0,
16.0, 
91.0, 
16.3, 
19.4, 
310.5, 
124.9, 
137.9, 
93.1, 
53.6, 
122.1, 
25.4, 
26.3,
196.9, 
68.8,
82.0, 
8.4, 
319.3, 
86.8, 
8.7, 
37.7, 
3.4, 
166.0,
104.2, 
45.0, 
14.5, 
38.1, 
17.8, 
31.1, 
89.0, 
66.3, 
17.7, 
4.5, 
69.1, 
1.1, 
6.2, 
47.2, 
23.4, 
160.8, 
6.4,
5.1, 
134.5, 4.1])

Redshift = array ('i', [0.1,
0.04,
 0.07,
 0.05,
 0.02,
 0.05,
 0.05,
 0.04,
 0.02,
 0.07,
0.0010,
0.12,
0.05,
0.04,
0.02,
0.0010,
0.09,
0.06,
0.02,
0.0010,
0.17,
0.02,
0.0010,
0.0010,
0.0010,
0.06,
0.03,
0.09,
0.0010,
0.03,
0.03,
0.04,
0.04,
0.06,
0.05,
0.0010,
0.05,
0.0010,
0.0010,
0.05,
0.0010,
0.06,
0.02,
0.02,
 0.03,
0.0010])
# Resizing the figure
plt.figure(figsize=[7, 5])

# Plotting the scatter plot
plt.scatter(x, y, c='g', alpha=0.6)
plt.title('Luminosity vs redshift', fontsize=15)
plt.xlabel('Redshift', fontsize=13)
plt.ylabel('Luminosity', fontsize=13)

# Changing to the log ticks at x and y axis using loglog
plt.loglog(basex=10, basey=2)

plt.show()